.github/workflows/build-and-test.yml

```diff
@@ -25,9 +25,11 @@ jobs:
       with:
         vs-version: '[17,18)'
         msbuild-architecture: x64
-    - name: Install libssl (Mac Only)
-      if: ${{ matrix.os == 'macos-11' }}
-      run: brew install openssl@3
+    - name: Install libssl and switch to XCode 15.2 (Mac Only)
+      if: ${{ matrix.os == 'macos-13' }}
+      run: |
+        brew install openssl@3
+        sudo xcode-select -s /Applications/Xcode_15.2.app/Contents/Developer
```

Comment:
Why is sudo being used here?  Is there no better way to accomplish the same work?

Reply:
The mode has to be switched manually, so I don't think there is a better way to do it.

Reply:
I don't have experience with xcode-select so I cant confirm or deny this.